src/os_crypto/shared/msgs.c

```diff
@@ -180,6 +184,43 @@ static void StoreCounter(const keystore *keys, int id, unsigned int global, unsi
     /* Write to the beginning of the file */
     fseek(keys->keyentries[id]->fp, 0, SEEK_SET);
     fprintf(keys->keyentries[id]->fp, "%u:%u:", global, local);
+    fflush(keys->keyentries[id]->fp);
+}
+
+/* Reload the global and local count of events */
+static void ReloadCounter(const keystore *keys, int id, const char * cid)
```

Comment:
This function does not update `global_count` or `local_count`. Those variables must be updated instead of the agent counters in the case that `id==keysize`.